bootspec/src/v1.rs

```diff
@@ -33,11 +31,13 @@ pub struct GenerationV1 {
     /// Path to the init script
     pub init: PathBuf,
     /// Path to initrd -- $toplevel/initrd
-    pub initrd: PathBuf,
+    pub initrd: Option<PathBuf>,
     /// Path to "append-initrd-secrets" script -- $toplevel/append-initrd-secrets
     pub initrd_secrets: Option<PathBuf>,
     /// Mapping of specialisation names to their boot.json
     pub specialisation: HashMap<SpecialisationName, GenerationV1>,
+    /// System double for the system closure
```

Comment:
This isn't really clear (to me) what it actually refers to. Could you maybe specify that this is specifically what Nix uses for system types? An example would also go far (e.g. mention `x86_64-linux` and `aarch64-linux`).